common/src/chain/block/consensus_data.rs

```diff
@@ -52,22 +51,28 @@ impl ConsensusData {
 }
 
 /// Fake PoS just to test spending block rewards; will be removed at some point in the future
-#[derive(Debug, Clone, PartialEq, PartialOrd, Ord, Eq, Encode, Decode)]
+#[derive(Debug, Clone, PartialEq, Eq, Encode, Decode)]
 pub struct PoSData {
     kernel_inputs: Vec<TxInput>,
     kernel_witness: Vec<InputWitness>,
+    stake_pool_id: PoolId,
+    vrf_data: VRFReturn,
     bits: Compact,
 }
 
 impl PoSData {
     pub fn new(
         kernel_inputs: Vec<TxInput>,
         kernel_witness: Vec<InputWitness>,
+        stake_pool_id: PoolId,
+        vrf_data: VRFReturn,
         bits: Compact,
     ) -> Self {
         Self {
             kernel_inputs,
             kernel_witness,
+            stake_pool_id,
```

Comment:
There's no need to manually put the pool id in the PoSData. The pool id can be derived from the block/transaction by reading its first input.

We can input the pool id when we want to stake, for example, because then we have to say "what output we're staking".

Reply:
Maybe I'm wrong here. Let's talk about it in chat.